app/src/main/java/com/umc/anddeul/home/PostWriteActivity.kt

```diff
@@ -51,20 +53,24 @@ class PostWriteActivity : AppCompatActivity() {
         // 이미지 URI 목록 받아오기
         val selectedImagesUri: ArrayList<Uri> = intent.getParcelableArrayListExtra("selectedImages")!!
 
-        val selectedVPAdapter = SelectedVPAdapter(selectedImagesUri)
+        selectedVPAdapter = SelectedVPAdapter(selectedImagesUri)
         binding.uploadWriteSelectedVp.adapter = selectedVPAdapter
         binding.uploadWriteSelectedVp.orientation = ViewPager2.ORIENTATION_HORIZONTAL
 
         // 받아온 이미지 URI 목록을 이용하여 이미지를 나열
-        val copiedImagesUri: List<Uri> = ArrayList(selectedImagesUri)
-        for (imageUri in copiedImagesUri) {
+        for (imageUri in selectedImagesUri) {
             selectedVPAdapter.addImage(imageUri)
         }
 
         binding.uploadWriteBtn.setOnClickListener {
             // 서버에 데이터 전송
             boardPost()
         }
+
+        // 뷰 페이저 드래그 앤 드랍 설정
+        val dragAndDropHelper = DragAndDropHelper(this)
```

Comment:
Activity 가 Listener interface 를 구현하는 방식은 예전 Java 스타일이에요!
지은님 40대 처럼 보이고싶지않으면 그냥 바로 생성해서 전달해주는게 나아요 ㅎㅎ
```suggestion
        val dragAndDropHelper = DragAndDropHelper(listener = object : DragAndDropHelper.OnItemMovedListener {
            override fun onItemMoved(fromPosition: Int, toPosition: Int) {
                selectedVPAdapter.swapItems(fromPosition, toPosition)
            }
        })
```

Reply:
아하 ㅎㅎㅎ Listener interface 없이 이렇게 전달하는 방법이 있었군요..!
앞으로는 이렇게 사용해보도록 하겠습니다 ㅎㅎ